python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_asset_evaluations.py

```diff
@@ -79,26 +81,40 @@ class Meta:
 
 
 def create_graphene_auto_materialize_condition(
-    condition_tuple: Tuple[AutoMaterializeCondition, Optional[SerializedPartitionsSubset]]
+    condition_tuple: Tuple[AutoMaterializeCondition, Optional[SerializedPartitionsSubset]],
+    partitions_def: Optional[PartitionsDefinition],
 ):
-    condition, _ = condition_tuple
+    condition, serialized_partition_subset = condition_tuple
+
+    if serialized_partition_subset and serialized_partition_subset.can_deserialize(partitions_def):
+        subset = serialized_partition_subset.deserialize(partitions_def)
```

Comment:
what's the expected behavior when the partitions def has changed and the partitions subset can no longer be deserialized?

Reply:
IMO, we should display a different message, something like "the partitions def has changed and the auto materialize condition can no longer be displayed".

Reply:
added an error message